app/models.py

```diff
@@ -182,85 +231,192 @@ def join_model(
+        """The Gravatar URL inferred from the user's contact email, or, if the
+        contact email is null, the username."""
+        email = (
+            self.contact_email
+            if self.contact_email is not None
+            else self.username
+        )
 
         digest = md5(email.lower().encode('utf-8')).hexdigest()
         # return a 404 status code if not found on gravatar
         return f'https://www.gravatar.com/avatar/{digest}?d=404'
 
     @property
     def acls(self):
+        """List of the user's ACLs."""
         return list({acl for role in self.roles for acl in role.acls})
 
     @property
     def permissions(self):
+        """List of the names of the user's ACLs."""
         return [acl.id for acl in self.acls]
 
     @classmethod
     def user_model(cls):
+        """The base model for User subclasses."""
         return User
 
     def is_authenticated(self):
+        """Boolean flag indicating whether the User is currently
+        authenticated."""
         return True
 
     def is_active(self):
+        """Boolean flag indicating whether the User is currently active."""
         return True
 
 
 class Token(Base):
+    """A command line token that can be used to programmatically access the API
+    as a particular User."""
+
     id = sa.Column(
-        sa.String, nullable=False, primary_key=True, default=lambda: str(uuid.uuid4())
+        sa.String,
+        nullable=False,
+        primary_key=True,
+        default=lambda: str(uuid.uuid4()),
+        doc="The value of the token. This field is used for authenticating as "
+        "a User on the command line.",
     )
+
     created_by_id = sa.Column(
-        sa.ForeignKey('users.id', ondelete='CASCADE'), nullable=True
+        sa.ForeignKey('users.id', ondelete='CASCADE'),
+        nullable=True,
+        doc="The ID of the User that created the Token.",
+    )
+    created_by = relationship(
+        'User', back_populates='tokens', doc="The User that created the token."
+    )
+    acls = relationship(
+        'ACL',
+        secondary='token_acls',
+        passive_deletes=True,
+        doc="The ACLs granted to the Token.",
+    )
+    acl_ids = association_proxy(
+        'acls',
+        'id',
+        creator=lambda acl: ACL.query.get(acl),
+        doc="The names of the ACLs granted to the token.",
     )
-    created_by = relationship('User', back_populates='tokens')
-    acls = relationship('ACL', secondary='token_acls', passive_deletes=True)
-    acl_ids = association_proxy('acls', 'id', creator=lambda acl: ACL.query.get(acl))
     permissions = acl_ids
     name = sa.Column(
-        sa.String, nullable=False, unique=True, default=lambda: str(uuid.uuid4())
+        sa.String,
+        nullable=False,
+        unique=True,
+        default=lambda: str(uuid.uuid4()),
+        doc="The name of the token.",
     )
 
     def is_owned_by(self, user_or_token):
+        """Return a boolean indicating whether a User (or Token) has read access
```

Comment:
This checks whether a token instance is owned by the specified user (or has the same identity as the specified token, if a token is passed in here)

Reply:
fixed